GroceryList/app/src/main/java/com/mobileproto/david/grocerylist/MainActivity.java

```diff
@@ -0,0 +1,156 @@
+package com.mobileproto.david.grocerylist;
+
+import android.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+
+/**
+ * This class is the main activity for our app.
+ */
+public class MainActivity extends AppCompatActivity
+        implements AdapterView.OnItemLongClickListener, View.OnClickListener
+{
+    static final String STATE_LIST_KEY = "list";
+    private static String logTag = "GroceryList";  // For debugging using Log.d(). Currently unused
+
+    private ArrayList<String> groceryItems;
+    private ArrayAdapter<String> itemsAdapter;
+
+    public ArrayAdapter<String> getItemsAdapter()
+    {
+        return itemsAdapter;
+    }
+
+    public ArrayList<String> getGroceryItems()
+    {
+        return groceryItems;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // If there is a savedInstanceState, load the groceryItems from the previous state.
+        if (savedInstanceState != null)
+            this.groceryItems = savedInstanceState.getStringArrayList(STATE_LIST_KEY);
```

Comment:
Interesting, I haven't seen this before. This looks like it will persist data if the app hasn't been killed?